ddraw/IDirect3DDeviceX.cpp

```diff
@@ -2125,6 +2125,12 @@ HRESULT m_IDirect3DDeviceX::DrawIndexedPrimitive(D3DPRIMITIVETYPE dptPrimitiveTy
 {
 	Logging::LogDebug() << __FUNCTION__ << " (" << this << ")";
 
+	if(dwIndexCount < 3)
```

Comment:
There are cases where the dwIndexCount can be less than 3.  For example if the dptPrimitiveType is D3DPT_POINTLIST or D3DPT_LINELIST, which use 1 or 2 indexes respectively.  It may be better if you change it to this:

```c++
if (GetNumberOfPrimitives(dptPrimitiveType, dwIndexCount) == 0)
```